specification/dartLangSpec.tex

```diff
@@ -13654,16 +13657,30 @@ \subsubsection{Ordinary Invocation}
 
 \LMHash{}%
 It is a compile-time error to invoke any of the methods of class \code{Object}
-on a constant type literal that is immediately followed by the token `.'\,.
+on a type literal that is immediately followed by the token `.' (a period).
 
 \rationale{%
-The reason for the latter is that this syntax is reserved for
-invocation of static methods.
+The reason for this rule is that this syntax is reserved for
+invocation of static members.
+Invocation of a static member of a class, mixin, or extension
+uses said entity as a \emph{namespace},
+not as an actual class, mixin, or extension.
+In particular, the syntactic receiver is not evaluated
+to an object---that would not even be possible for an extension.%
+}
+
+\commentary{%
 For instance, \code{int.toString()} is similar to
-\code{C.someStaticMethod()}, and
-it would be confusing if just a couple of expressions of this form
-were instance method invocations.
-If needed, \code{(int).toString()} may be used instead.%
+\code{C.someStaticMethod()}, and it would be confusing
+if the former would evaluate the receiver and call an instance method,
+and the latter (which is much more common and useful)
+would use the type literal as a namespace.
+If needed, \code{(int).toString()} may be used instead.
```

Comment:
This also confuses me a little, rather than illuminating me.
In general, don't talk about hypotheticals that we could have done instead, and then explain why we didn't. That's introducing more concepts for the reader to understand, without actually helping them understanding *what is*.

Maybe just make it short:

> An member invocation or property access on a type literal, `L.id` or `L?.id`, 
> always treas the declaration denoted by the literal as a namespace
> for accessing static members.
> For instance, `int.toString()` will be an error because `int`
> does not declare a static `toString` member.
> It will not evaluate `int` to a `Type` object and then call its `toString` instance method,
> to do that, you should write `(int).toString()` instead.
> (Cascades are different, though, since they always evaluate their receiver to an object first.).

Reply:
Yes, that's better. Done.